src/kloewe/dot/src/dot_sse2.h

```diff
@@ -48,17 +41,17 @@ inline double sddot_sse2 (const float  *a, const float  *b, int n);
 inline float sdot_sse2 (const float *a, const float *b, int n)
 {
   // initialize total sum
-  float s = 0.0;
+  float s = 0.0f;
 
-  // compute and add up to 3 products without SIMD, hoping for alignment
+  // compute and add up to 3 products without SIMD to achieve alignment
   int aligned = is_aligned(a, 16) && is_aligned(b, 16);
   if (!aligned) {
     int k = 0;
     while (!aligned) {
       s += (*a) * (*b);
       n--; a++; b++;
       aligned = is_aligned(a, 16) && is_aligned(b, 16);
-      if (aligned || (++k > 2) || (++k > n))
+      if (aligned || (++k > 2) || (k > n))
```

Comment:
n has been decremented, and k incremented, so instead of ```(k > n)```, it should probably be ```(n > 0)```.  Ditto for the other implementations.

Reply:
Thanks, good catch! You meant `(n == 0)`, though, right?

Reply:
Ah, right...its always something.